todo_app/app.py

```diff
@@ -1,102 +1,43 @@
 from todo_app.view_model import ViewModel
+from todo_app.todomongo import add_todo_item, items, update_status, delete_item
 
 def create_app():
     app = Flask(__name__)
     app.config.from_object(Config())
 
 
     @app.route('/')
-    def index():
-        
-        url = f"https://api.trello.com/1/boards/{os.getenv('TRELLO_BOARD_ID')}/lists"
-
-        querystring = {
-                "key":os.getenv("TRELLO_API_KEY"),
-                "token":os.getenv("TRELLO_API_TOKEN"),
-                "cards": "open"
-        }
 
-        response = requests.get(url, params=querystring)
-
-        response_json = response.json()
-
-        items = []
-
-        for trello_list in response_json:
-            for card in trello_list['cards']:
-                myItem = Item(card['id'] , card['name'] , trello_list['name'])
-                items.append(myItem)
+    def index():
 
-        item_view_model = ViewModel(items)
+        item_view_model = ViewModel(items())
         return render_template('index.html',
         view_model=item_view_model)
 
-    @app.route('/add_card', methods=['POST'])
-    def add_new_card():
-
-        url = "https://api.trello.com/1/cards"
+    @app.route('/add_task', methods=['POST'])
+    def add_new_item():
 
         title = request.form['todo_title']
+        add_todo_item(title)
+        
+        return redirect ('/')
 
-        querystring = {
-        "key":os.getenv("TRELLO_API_KEY"),
-        "token":os.getenv("TRELLO_API_TOKEN"),
-        "idList" : os.getenv("TRELLO_LIST_TODO"),
-        "name" : title
-
-        }
-
-        response = requests.request("POST", url, params=querystring)
+    @app.route('/progress', methods=['POST'])
+    def in_progress():
 
-        card_id = response.json()
-        
-        return index ()
+        item_id = request.form['item_id']
+        update_status(item_id, 'in progress')
+  
+        return redirect ('/')
         
     @app.route('/mark_done', methods=['POST'])
     def mark_done():
 
-        card_id = request.form['card_id']
-    
-        url = f"https://api.trello.com/1/cards/{card_id}"
-
-        querystring = {
-        "key":os.getenv("TRELLO_API_KEY"),
-        "token":os.getenv("TRELLO_API_TOKEN"),
-        "idList" : os.getenv("TRELLO_LIST_DONE"),
-
-        }
-
-        response = requests.request("PUT", url, params=querystring)
-
-        card_id = response.json()
-
-        
-        return index ()
-
-    @app.route('/in_progress', methods=['POST'])
-    def in_progress():
-
-        card_id = request.form['card_id']
-    
-        url = f"https://api.trello.com/1/cards/{card_id}"
-
-        querystring = {
-        "key":os.getenv("TRELLO_API_KEY"),
-        "token":os.getenv("TRELLO_API_TOKEN"),
-        "idList" : os.getenv("TRELLO_LIST_IN_PROGRESS"),
-
-        }
-
-        response = requests.request("PUT", url, params=querystring)
-
-        card_id = response.json()
-
+        item_id = request.form['item_id']
+        delete_item(item_id)
```

Comment:
Was it intentional to move this from a route/method that marked something complete to one that deletes it? Potentially a subtle change in behaviour

Reply:
It was intentional, I thought it would be better to delete it off rather than have lots of completed tasks floating about.